### ISSUE TITLE
wrong nested `package.json` resolved

### ISSUE THREAD

**Issue by JounQin** [xojs/xo#741]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `xo@0.56.0` for the project I'm working on.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/xo/lib/options-manager.js b/node_modules/xo/lib/options-manager.js
index 6feb6ec..1d6db87 100644
--- a/node_modules/xo/lib/options-manager.js
+++ b/node_modules/xo/lib/options-manager.js
@@ -108,7 +108,11 @@ const mergeWithFileConfig = async options => {
 	}
 
 	const searchPath = options.filePath || options.cwd;
-	const {config: xoOptions, filepath: xoConfigPath} = (await configExplorer.search(searchPath)) || {};
+	let {config: xoOptions, filepath: xoConfigPath} = (await configExplorer.search(searchPath)) || {};
+  // wrong nested `package.json` resolved
+  if (!xoOptions && xoConfigPath) {
+    ({config: xoOptions, filepath: xoConfigPath} = (await configExplorer.search(path.resolve(xoConfigPath, '../..'))) || {});
+  }
 	const {config: enginesOptions} = (await pkgConfigExplorer.search(searchPath)) || {};
 
 	options = normalizeOptions({
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by JounQin** [comment]
Working on a PR.

**Comment by spence-s** [comment]
Could you describe the scenario better? I don't think this fix is what we want to do. 

I take it that you are in a mono-repo, you have a xo installed in a nested package, and your xo config is at the root level in your package.json.

.
├── packages
   └── some-package
         |------- package.json (xo installed here, but config is 2 dirs up)
|- package.json (xo config here)

You should instead do something like 
from your nested package  folder:
`npx xo --cwd ../..`

This will fix your problem I think, it's a common trip up for people with mono-repos.

**Comment by JounQin** [comment]
> `npx xo --cwd ../..`

It seems too verbose to me personally, and I didn't know it would work. Why not fix like this?

**Comment by JounQin** [comment]
OK, I found out that it's actually specifically disabled at https://github.com/rehypejs/rehype-github/blob/7d7e851b751413c1ec95e3fbdf1c48e4b71155e5/packages/break/package.json#L37, however in this case, `xo` should just ignore linting actually or change to use the root config like the current fix.

**Comment by JounQin** [comment]
> Because previously xo used xo: false for nested packages, to continue looking for a parent package.json.
> Omitting xo: false, previously, resulted in the defaults being applied.
> Which is exactly what xo is now doing when xo: false is used.

I think I'll change the fix by check whether `xo` is `false` instead. Do you think this is a regression? @spence-s

**Comment by spence-s** [comment]
At some point, years ago, xo would search all the way up to the home directory for a "global" xo config. But this in itself caused lots of confusion and bugs for many users and was hard to integrate with eslint properly. (ie, couldn't lookup plugins and configs correctly for configs outside of the repo)...

This behavior was specifically removed in favor of requiring configs to live in the repos, so this is not a bug nor a regression, but is working as intended and stopping search at `cwd` as intended.

_Does it still limit people with mono-repos?_ Yes it does limit them a bit and they may have to "hack" cwd like I showed above.

_Can you still use xo with mono repos using the recommended approach?_ absolutely, xo works just fine in mono-repos if set up correctly.

Could we improve this or accept functionality that improves this UX? That would be up to @sindresorhus, but probably would be willing to change things slightly if there was a good argument for it. 

However, it looks to me that xo is just being used in a mono repo set up here in a way that doesn't play nicely with its design. I would recommend refactoring the original repo per recommended mono repo setup.

TL;DR: not a bug nor regression, nor do the PRs that keep searching up something we want to add at this time. If/when we move to the the new config style, we may reconsider if that makes solving resolution bugs simpler.

**Comment by spence-s** [comment]
I would personally not support this behavior change. 

You can already get the behavior you want from the way it is _right now_. 

A pr for this would need to be comprehensive and have a lot of tests to account for resolving plugins and configs potentially outside of the repo.

Note: if you have an `xo` field - even if it is "false" in the package.json, search will prematurely stop there. You need to remove the xo field entirely from package.json to have it continue searching up to the cwd you set.

**Comment by wooorm** [comment]
`xo: false` was recommended before. I don’t think this feature was removed that many years ago (https://github.com/xojs/xo/commit/e0f81a724dfc487babced845067e9d49d630c56f).
I’m pretty sure we’ve used it in many places, since it was added 7 years ago, up until maybe a year ago? 50 or so?

Anyway, I’m fine not supporting this.

**Comment by spence-s** [comment]
The PR you linked was from 3.5 years ago and it does look like the change happened there.

It also includes the documentation to simply omit the "xo" field in order to continue searching up to the cwd, changed from the previously recommended `xo: false` 

I would support this change to respect `xo: false` as long as it definitely stops searching at the cwd (like it does now), which would be a very minor behavior change.

However, its probably easier to just remove the xo: false fields from your package.jsons 

as with all changes the final say is up to @sindresorhus not me.

**Comment by sindresorhus** [comment]
I don't plan to bring back `xo: false`. The good news is that all plugins and stuff is no in `eslint-config-xo`, so you can just use ESLint directly.
